src/HTTPRequest.cpp

```diff
@@ -60,23 +61,42 @@ void HTTPRequest::post(const URL& url,
 void HTTPRequest::get(const URL& url,
                       std::function<void(const std::string&)> onSuccess,
                       std::function<void(const std::string&)> onError,
-                      const std::string& fileName)
+                      const std::string& fileName,
+                      const unsigned int retryAttempts)
 {
-    try
+    std::string exceptionMessage;
+    unsigned int attempts {1 + retryAttempts};
+
+    // Try the request 'attempts' times
+    while (0 < attempts)
     {
         auto req {GetRequest::builder(FactoryRequestWrapper<wrapperType>::create())};
-        req.url(url.url())
-            .appendHeader("Content-Type: application/json")
-            .appendHeader("Accept: application/json")
-            .appendHeader("Accept-Charset: utf-8")
-            .outputFile(fileName)
-            .execute();
+
+        try
+        {
+            req.url(url.url())
+                .appendHeader("Content-Type: application/json")
+                .appendHeader("Accept: application/json")
+                .appendHeader("Accept-Charset: utf-8")
+                .outputFile(fileName)
+                .execute();
+        }
+        catch (const std::exception& ex)
+        {
+            attempts--;
```

Comment:
Pre decrements have fewer instructions.
```suggestion
            --attempts;
```